### PR TITLE
Messaging API

### PR DESCRIPTION
**Pull request by jonathanguven**
Frontend for the Messaging API Platform

1. Form for sending messages
2. Live feed for default room

### CODE REVIEW

File: src/APIFunctions/Messaging.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    })
+    .catch(err => {
+      status.error = true;
+      status.responseData = err;
+    });
+  return status;
+}
+
+export async function connectToRoom(room, apiKey, onMessage, onError) {
+  const eventSource = new EventSource(`${GENERAL_API_URL}/messages/listen?id=${encodeURIComponent(room)}&apiKey=${encodeURIComponent(apiKey)}`);
```
Comment (evanugarte) [inline src/APIFunctions/Messaging.js]: can we do this, also on line 11 too
```suggestion
  const url = new URL(`${GENERAL_API_URL}/messages/listen`);
  url.searchParams.append('id', room);
  url.searchParams.append('apiKey', apiKey);
  const eventSource = new EventSource(url.href);
```

File: src/APIFunctions/Messaging.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  eventSource.onmessage = (event) => {
+    const data = JSON.parse(event.data);
```
Comment (evanugarte) [inline src/APIFunctions/Messaging.js]: add a try/catch in here? or will `eventSource.onerror` handle paring errors for us?

File: src/Pages/Messaging/Messaging.js [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+function Feed(props) {
+  const { token } = props;
+  const [messages, setMessages] = useState([]);
+  const [apiKey, setApiKey] = useState('');
+  const [error, setError] = useState('');
+  const roomId = 'general';
+  let eventSource;
+
+  useEffect(() => {
+    const fetchApiKey = async () => {
```
Comment (evanugarte) [inline src/Pages/Messaging/Messaging.js]: how about for sending messages, we allow the user to pass in a token, that way we dont need to handle api key logic in here


api keys can be the second option that our messaging apis use, if a token wasnt sent, but an api key was, let me know if this makes sense

by allowing tokens, we dont need this function

File: src/APIFunctions/Messaging.js (hunk 2 above)
Comment (flotoria) [reply src/APIFunctions/Messaging.js]: i think ```eventSource.onerror``` should handle the errors

File: src/Pages/Messaging/Messaging.js (hunk 3 above)
Comment (flotoria) [reply src/Pages/Messaging/Messaging.js]: lemme handle this

Comment (flotoria) [reply src/Pages/Messaging/Messaging.js]: implemented, probably need to check whether the solution is clean or not

**Review by akshtshrma24 — APPROVED** [review]

**Review by akshtshrma24 — APPROVED** [review]

**Review by akshtshrma24 — APPROVED** [review]

**Review by flotoria — APPROVED** [review]

**Review by evanugarte — APPROVED** [review]

**Review by food0903 — APPROVED** [review]

**Pull request merged**